src/setup/node.rs

```diff
@@ -171,16 +141,72 @@ impl Drop for Node {
     }
 }
 
+/// Convenience struct to better control configuration/build/start for [Node]
+pub struct NodeBuilder {
+    config: NodeConfig,
+    meta: NodeMetaData,
+}
+
+impl NodeBuilder {
+    /// Sets up minimal configuration for the node.
+    pub fn new(path: PathBuf, ip_addr: IpAddr) -> Result<Self> {
+        let config = NodeConfig::new(path, ip_addr);
+        let meta = NodeMetaData::new(config.path.clone())?;
+        Ok(Self { config, meta })
+    }
+
+    /// Sets initial peers for the node.
+    pub fn initial_peers(mut self, addrs: Vec<SocketAddr>) -> Self {
+        self.config.initial_peers = addrs.into_iter().collect();
+        self
+    }
+
+    /// Sets validator token to be placed in rippled.cfg.
+    /// This will make the node to be run in validator mode.
+    pub fn validator_token(mut self, token: String) -> Self {
+        self.config.validator_token = Some(token);
+        self
+    }
+
+    /// Sets whether to log the node's output to Ziggurat's output stream.
+    pub fn log_to_stdout(mut self, log_to_stdout: bool) -> Self {
+        self.config.log_to_stdout = log_to_stdout;
+        self
+    }
+
+    /// Builds and starts the new node.
+    pub async fn build(self) -> Result<Node> {
+        let mut node = Node {
+            config: self.config,
+            meta: self.meta,
+            process: None,
+        };
+        node.start_process().await?;
+        Ok(node)
+    }
+}
+
 #[cfg(test)]
 mod tests {
+    use std::net::Ipv4Addr;
+
     use super::*;
+    use crate::setup::config::ZIGGURAT_CONFIG_DIR;
 
     #[ignore = "convenience test to tinker with a running node for dev purposes"]
     #[tokio::test]
     async fn start_stop_node() {
-        let mut node = Node::new().unwrap();
-
-        node.log_to_stdout(true).start().await.unwrap();
+        let mut node = NodeBuilder::new(
```

Comment:
I wrote this test when initially bootstrapping the repo, do we still need it?

Reply:
Not necessarily.  Should I remove it?

Reply:
Sure, if there's no need, let's cut down on the code we need to maintain 👍